BraveWallet/Panels/Signature Request/SignatureRequestView.swift

```diff
@@ -126,26 +148,52 @@ struct SignatureRequestView: View {
   }
   
   @ViewBuilder private var buttons: some View {
-//    Button(action: {
-//
-//    }) {
-//      Text(Strings.cancelButtonTitle)
-//    }
-//    .buttonStyle(BraveOutlineButtonStyle(size: .large))
-    Button(action: {
-      
+    Button(action: { // cancel
+      cryptoStore.handleWebpageRequestResponse(.signMessage(approved: false, id: currentRequest.id))
+      if requests.count > 1 {
+        requests.removeFirst()
+      } else {
+        onDismiss()
+      }
     }) {
-      Label("Sign", image: "brave.key")
+      Label(Strings.cancelButtonTitle, systemImage: "xmark")
+        .imageScale(.large)
+    }
+    .buttonStyle(BraveOutlineButtonStyle(size: .large))
+    .disabled(isButtonsDisabled)
+    Button(action: { // approve
+      cryptoStore.handleWebpageRequestResponse(.signMessage(approved: true, id: currentRequest.id))
+      if requests.count > 1 {
+        requests.removeFirst()
```

Comment:
This should _probably_ remove the request in crypto store side and re-push into this view rather than remove here and abuse `State` initialization